sh/configure_env.sh

```diff
@@ -0,0 +1,106 @@
+#
+# Author: Philip Deegan
+# Email : philip.deegan@polytechnique.edu 
+# Date  : 21 - September - 2017
+#
+# This script is to be included to setup variables to build tick
+#   It is done so to be cross platform and heterogeneous
+#
+# Variables
+#  $PY = "python" binary
+#  $PCONF = "python-config" binary
+#  $LDARGS=Linking arguments
+#  $LIB_POSTFIX= python library format 
+#  $LIB_POSTEXT= system library file extention
+#  $PROFILES= array of mkn profiles to build
+# 
+######################################################################
+
+set -e
+
+[ -z "$PY" ] && which python3 &> /dev/null && PY="python3"
+[ -z "$PY" ] && which python &> /dev/null && PY="python"
+[ -z "$PY" ] && echo "python or python3 not on path, exiting with error" && exit 1
+
+[ -z "$PCONF"  ] && which python3-config &> /dev/null && PCONF="python3-config"
+[ -z "$PCONF"  ] && which python-config &> /dev/null && PCONF="python-config"
+[ -z "$PCONF"  ] && echo "python-config or python3-config not found on PATH, exiting with error" && exit 1;
+
+[ -z "$PINC" ] && for i in $($PCONF --includes); do PINC="${PINC}${i:2}:"; done
+[ -z "$PNIC" ] && PNIC=$($PY -c "import numpy as np; print(np.get_include())");
+
+LDARGS="$($PCONF --ldflags)"
+LDARGS=$(echo "$LDARGS" | sed -e "s/ -lintl//g")
+LDARGS=$(echo "$LDARGS" | sed -e "s/ -ldl//g")
+LDARGS=$(echo "$LDARGS" | sed -e "s/ -framework//g")
+LDARGS=$(echo "$LDARGS" | sed -e "s/ CoreFoundation//g")
+
+unameOut="$(uname -s)"
+case "${unameOut}" in
+    Linux*)     LDARGS="${LDARGS} $(mkn -G nix_largs)";;
+    Darwin*)    LDARGS="${LDARGS} $(mkn -G bsd_largs)";;
+    CYGWIN*)    LDARGS="${LDARGS}";;
+    MINGW*)     LDARGS="${LDARGS}";;
+    *)          LDARGS="${LDARGS}";;
+esac
+[ ! -z "$LDFLAGS" ] && LDARGS="${LDARGS} ${LDFLAGS}"
+
+# IFS = Internal Field Separator - allows looping over lines with spaces
+IFS=$'\n'
```

Comment:
Would this even work on Windows? I thought window was using \r\n but it's something I don't know much about.

Reply:
Windows still requires MSYS or CYGWIN to use this, so it should be ok

Reply:
Oh yes right!